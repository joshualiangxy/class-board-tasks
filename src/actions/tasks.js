import firebase from '../firebase/firebase';

export const addPersonalTask = task => ({ type: 'ADD_PERSONAL_TASK', task });

export const startAddPersonalTask = task => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const id = task.id;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(task)
      .then(() => dispatch(addPersonalTask(task)));
  };
};

export const removePersonalTask = id => ({ type: 'REMOVE_PERSONAL_TASK', id });

export const startRemovePersonalTask = id => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .delete()
      .then(() => dispatch(removePersonalTask(id)));
  };
};

export const editPersonalTask = (id, updates) => ({
  type: 'EDIT_PERSONAL_TASK',
  id,
  updates
});

export const startEditPersonalTask = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(updates)
      .then(dispatch(editPersonalTask(id, updates)));
  };
};

export const setTasks = tasks => ({ type: 'SET_TASK', tasks });

export const startSetTasks = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const tasks = [];
    const userRef = firebase.firestore().collection('users').doc(uid);
    const tasksRef = userRef.collection('tasks');

    return tasksRef
      .get()
      .then(querySnapshot =>
        querySnapshot.forEach(queryDocSnapshot =>
          tasks.push({ id: queryDocSnapshot.id, ...queryDocSnapshot.data() })
        )
      )
      .then(() => {
        userRef.get().then(userDocSnapshot => {
          const groupIds = userDocSnapshot.get('groups') || [];
          const groupsPromises = [];

          groupIds.forEach(gid => {
            const groupPromises = [];
            const groupRef = firebase.firestore().collection('groups').doc(gid);

            groupPromises.push(
              groupRef
                .get()
                .then(groupDocSnapshot => groupDocSnapshot.get('name'))
            );
            groupPromises.push(groupRef.collection('tasks').get());

            groupsPromises.push(
              Promise.all(groupPromises).then(
                ([groupName, queryTaskCollectionSnapshot]) =>
                  queryTaskCollectionSnapshot.forEach(queryTaskDocSnapshot =>
                    tasks.push({
                      groupName,
                      gid,
                      id: queryTaskDocSnapshot.id,
                      ...queryTaskDocSnapshot.data()
                    })
                  )
              )
            );

            return Promise.all(groupPromises);
          });
        });
      })
      .then(() => dispatch(setTasks(tasks)));
  };
};

export const toggleCompletedPersonal = (id, completedState) => ({
  type: 'TOGGLE_COMPLETED_PERSONAL',
  id,
  completedState
});

export const startToggleCompletedPersonal = (id, completedState) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .update({ completed: !completedState })
      .then(() => toggleCompletedPersonal(id, completedState));
  };
};

export const removeTaskData = () => ({ type: 'REMOVE_TASK_DATA' });
