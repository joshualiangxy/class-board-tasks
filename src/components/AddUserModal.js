import React, { useState } from 'react';
import Modal from 'react-modal';
import firebase from '../firebase/firebase';

// Note: This still works off uid for now. This also does NOT work yet.
const AddUserModal = ({ isOpen, onRequestClose, group }) => {
  const addNewUser = (userEmail, group) => {
    const uid = userEmail; // TODO: Change this when email-uid are linked!

    // Both of these references are DocumentReference
    const userRef = firebase.firestore().collection('users').doc(uid);
    const groupRef = firebase.firestore().collection('groups').doc(group);

    // Gets the data from a user.
    userRef.get().then(doc => {
      const data = doc.data();
      // Write displayName, studentNum, and UID into a new document
      if (typeof data === 'undefined') {
        // Most likely a user that doesn't exist.
        setError('No such user found');
      } else {
        groupRef
          .collection('users')
          .doc(uid) // Create document for this user's UID
          .set({
            studentNum: data.studentNum,
            displayName: data.displayName,
            uid: uid,
            admin: false
          })
          .then(() => {
            setError('User added!');
          })
          .catch(error => {
            setError('Something went wrong.');
          });
      }
      // Defaults to non-admin.
    });
  };

  // TODO: When calling this modal, pass in the group UID as a prop
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState('');
  // const [usersAdded, setUsersAdded] = useState([]); // For listing down people that have been added?

  const onUserEmailChange = e => {
    setUserEmail(e.target.value);
  };

  const onCancel = () => {
    setUserEmail('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedEmail = userEmail.trim();
    if (!submittedEmail) {
      setError('Please enter a group name');
    } else {
      addNewUser(submittedEmail, group);
      // TODO: Catch user not found
      // Does not close after submitting, so that the user can continue to add more users

      // setUserEmail(''); // Also resets the field.
      // setError('Added!');
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={() => onRequestClose()}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="User ID (required)"
          value={userEmail}
          onChange={onUserEmailChange}
          autoFocus
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

export default AddUserModal;
