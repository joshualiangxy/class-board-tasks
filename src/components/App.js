import React from 'react';
import '../styles/App.css';

const App = () => (
  <div>
    <h1>This is my app</h1>
  </div>
);

export default App;

