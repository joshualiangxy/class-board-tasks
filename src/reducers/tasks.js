const tasksReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TASK':
      return [...state, action.task];
    case 'REMOVE_TASK':
      return state.filter(task => task.id !== action.id);
    case 'EDIT_TASK':
      return state.map(task =>
        task.id === action.id
          ? {
              ...task,
              ...action.updates
            }
          : task
      );
    case 'SET_TASK':
      return action.tasks;
    default:
      return state;
  }
};

export default tasksReducer;
