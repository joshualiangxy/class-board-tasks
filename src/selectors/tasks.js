const getSortedTasks = (tasks, { text, sortBy, grouped }) => {
  const lowercaseText = text.toLowerCase();

  const processedTasks = tasks
    .filter(
      task =>
        task.title.toLowerCase().includes(lowercaseText) ||
        task.description.toLowerCase().includes(lowercaseText) ||
        task.module.toLowerCase().includes(lowercaseText)
    )
    .sort((taskOne, taskTwo) => {
      switch (sortBy) {
        case 'deadline':
          if (taskOne.deadline && taskTwo.deadline)
            return taskOne.deadline - taskTwo.deadline;
          else if (taskOne.deadline) return -1;
          else if (taskTwo.deadline) return 1;
          else return taskOne.title.localeCompare(taskTwo.title);
        case 'name':
          return taskOne.title.localeCompare(taskTwo.title);
      }
    });

  const sortedList = {};

  if (grouped) {
    processedTasks.forEach(task => {
      if (task.groupName) {
        const gid = task.gid;

        if (!sortedList[gid])
          sortedList[gid] = { groupName: task.groupName, tasks: [] };
        sortedList[gid].tasks.push(task);
      } else if (task.module) {
        const module = task.module;

        if (!sortedList[module])
          sortedList[module] = { groupName: task.module, tasks: [] };
        sortedList[module].tasks.push(task);
      } else {
        if (!sortedList.others)
          sortedList.others = { groupName: 'others', tasks: [] };
        sortedList.others.tasks.push(task);
      }
    });
  } else sortedList.others = { groupName: 'others', tasks: processedTasks };

  return sortedList;
};

export default getSortedTasks;
