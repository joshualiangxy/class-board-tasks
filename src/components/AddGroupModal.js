import React, { useState } from 'react';
import Modal from 'react-modal';
import { startNewGroup } from '../actions/groups';
import { connect } from 'react-redux';

// TODO: Add the ability to add other users into the group
const AddGroupModal = ({ isOpen, onRequestClose, startNewGroup }) => {
  const [groupName, setGroupName] = useState('');
  const [module, setModule] = useState('');
  const [error, setError] = useState('');

  const onGroupNameChange = e => setGroupName(e.target.value);

  const onModuleChange = e => {
    const module = e.target.value.toUpperCase();

    setModule(module);
  };

  const onCancel = () => {
    setError('');
    setGroupName('');
    setModule('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedName = groupName.trim();
    const submittedMod = module.trim();
    if (!submittedName) {
      setError('Please enter a group name');
    } else {
      setError('');
      setGroupName('');
      setModule('');
      startNewGroup(submittedName, submittedMod).then(() => onRequestClose());
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={onCancel}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Group name (required)"
          value={groupName}
          onChange={onGroupNameChange}
          autoFocus
        />
        <input
          type="text"
          placeholder="Module code/Class name (Optional)"
          value={module}
          onChange={onModuleChange}
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startNewGroup: (groupName, module) =>
    dispatch(startNewGroup(groupName, module))
});

export default connect(null, mapDispatchToProps)(AddGroupModal);
