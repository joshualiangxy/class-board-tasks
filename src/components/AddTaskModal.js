import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startAddPersonalTask, startAddGroupTask } from '../actions/tasks';
import TaskForm from './TaskForm';

export const AddTaskModal = ({
  startAddPersonalTask,
  startAddGroupTask,
  removeAdmin,
  isOpen,
  onRequestClose,
  gid,
  groupModule,
  groupName,
  addGroupTask,
  users
}) => {
  const submitTask = task =>
    gid
      ? startAddGroupTask(task, groupName)
          .then(isAdmin => {
            if (isAdmin) return addGroupTask(task);
            else return removeAdmin();
          })
          .then(() => onRequestClose())
      : startAddPersonalTask(task).then(() => onRequestClose());

  return (
    <Modal
      className="task-modal"
      isOpen={isOpen}
      contentLabel="Add Task"
      onRequestClose={onRequestClose}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <h1>Add Task</h1>
      <TaskForm
        gid={gid}
        groupModule={groupModule}
        users={users}
        submitTask={submitTask}
        onRequestClose={onRequestClose}
      />
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startAddPersonalTask: task => dispatch(startAddPersonalTask(task)),
  startAddGroupTask: (task, groupName) =>
    dispatch(startAddGroupTask(task, groupName))
});

export default connect(undefined, mapDispatchToProps)(AddTaskModal);
