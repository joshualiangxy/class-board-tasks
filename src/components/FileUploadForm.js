import React, { useState } from 'react';
import firebase from '../firebase/firebase';
import { startUpdateDownloadURL } from '../actions/tasks';
import { uploadFile, startRemoveUserFile } from '../actions/files';
import { connect } from 'react-redux';

export const FileUploadForm = ({
  id,
  uid,
  uploadFile,
  TaskEvent,
  startUpdateDownloadURL,
  startRemoveUserFile,
  fileExists
}) => {
  const [error, setError] = useState('');
  const [uploading, setUploading] = useState(false);

  const fileUpload = e => {
    e.preventDefault();
    e.persist();

    const element = document.getElementById(id);
    const files = element.files;
    if (files.length > 0) {
      const file = files[0];

      Promise.resolve()
        .then(() => {
          if (fileExists) return startRemoveUserFile(uid);
        })
        .then(() => {
          const uploadTask = uploadFile(file);

          uploadTask.on(
            TaskEvent.STATE_CHANGED,
            () => setUploading(true),
            error => {
              switch (error.code) {
                case 'storage/unauthorized':
                  setError('Not allowed to upload file!');
                  break;
                case 'storage/canceled':
                  setError('File upload cancelled');
                  break;
                default:
                  setError('File upload failed');
                  break;
              }
              setUploading(false);
            },
            () => {
              const ref = uploadTask.snapshot.ref;
              const fileName = ref.name;

              ref
                .getDownloadURL()
                .then(downloadURL => {
                  startUpdateDownloadURL(downloadURL, fileName);
                })
                .then(() => {
                  setError('');
                  setUploading(false);
                  e.target.reset();
                });
            }
          );
        });
    } else setError('No file selected');
  };

  return (
    <form onSubmit={fileUpload}>
      {error && <p>{error}</p>}
      <input
        type="file"
        id={id}
        onClick={e => e.stopPropagation()}
        accept=".pdf"
        disabled={uploading}
      />
      <button onClick={e => e.stopPropagation()} disabled={uploading}>
        {uploading ? 'Uploading...' : 'Upload'}
      </button>
    </form>
  );
};

const mapStateToProps = ({ tasks, auth }, { id }) => {
  const uid = auth.user.uid;
  const task = tasks.find(task => task.id === id);

  return {
    TaskEvent: firebase.storage.TaskEvent,
    fileExists: task ? task.downloadURLs.hasOwnProperty(uid) : false,
    uid
  };
};

const mapDispatchToProps = (dispatch, { id, gid, namingConvention }) => ({
  uploadFile: file => dispatch(uploadFile(file, id, namingConvention)),
  startUpdateDownloadURL: (downloadURL, fileName) =>
    dispatch(startUpdateDownloadURL(id, gid, downloadURL, fileName)),
  startRemoveUserFile: uid => dispatch(startRemoveUserFile(id, uid))
});

export default connect(mapStateToProps, mapDispatchToProps)(FileUploadForm);
