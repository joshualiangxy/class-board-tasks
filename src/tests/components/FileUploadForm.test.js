import React from 'react';
import { JSDOM } from 'jsdom';
import { shallow } from 'enzyme';
import { FileUploadForm } from '../../components/FileUploadForm';
import { groupTasks } from '../fixtures/tasks';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

let wrapper;
let files = ['file'];
const groupTask = groupTasks[2];
const id = groupTask.id;
const uid = 'testuid';
const TaskEvent = { STATE_CHANGED: 'STATE_CHANGED' };
const startRemoveUserFile = jest.fn();
const startUpdateDownloadURL = jest.fn();
const getDownloadURL = jest.fn(() => Promise.resolve('www.download.com'));
const uploadTask = {
  on: jest.fn(),
  snapshot: { ref: { name: 'fileName', getDownloadURL } }
};
const uploadFile = jest.fn(() => uploadTask);
const element = { files };
const submitEvent = {
  preventDefault: jest.fn(),
  persist: jest.fn(),
  target: { reset: jest.fn() }
};

beforeEach(() => {
  jest.clearAllMocks();
  uploadTask.on.mockImplementation((state, observer, error, callback) =>
    Promise.resolve(observer()).then(() => callback())
  );
  global.document.getElementById = jest.fn(() => element);

  files = ['file'];
  wrapper = shallow(
    <FileUploadForm
      id={id}
      uid={uid}
      uploadFile={uploadFile}
      TaskEvent={TaskEvent}
      startUpdateDownloadURL={startUpdateDownloadURL}
      startRemoveUserFile={startRemoveUserFile}
      fileExists={false}
    />
  );
});

describe('render', () => {
  it('should render FileUploadForm', () => expect(wrapper).toMatchSnapshot());
});

describe('error', () => {
  it('should render unauthorized error', () => {
    uploadTask.on.mockImplementation((state, observer, error, callback) =>
      Promise.resolve(observer()).then(() =>
        error({ code: 'storage/unauthorized' })
      )
    );

    return wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => expect(wrapper).toMatchSnapshot());
  });

  it('should render cancelled error', () => {
    uploadTask.on.mockImplementation((state, observer, error, callback) =>
      Promise.resolve(observer()).then(() =>
        error({ code: 'storage/canceled' })
      )
    );

    return wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => expect(wrapper).toMatchSnapshot());
  });

  it('should render default error', () => {
    uploadTask.on.mockImplementation((state, observer, error, callback) =>
      Promise.resolve(observer()).then(() => error({ code: 'default' }))
    );

    return wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => expect(wrapper).toMatchSnapshot());
  });

  it('should render error when no file is selected', () => {
    files.pop();

    wrapper.find('form').prop('onSubmit')(submitEvent);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('remove existing file', () => {
  it('should remove existing file if file already exists', () => {
    wrapper.setProps({ fileExists: true });

    return wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => {
        expect(startRemoveUserFile).toHaveBeenCalledTimes(1);
        expect(startRemoveUserFile).toHaveBeenLastCalledWith(uid);
      });
  });

  it('should not remove any files if file does not exist', () =>
    wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => expect(startRemoveUserFile).toHaveBeenCalledTimes(0)));
});

describe('upload form', () => {
  it('should change button to "Uploading..." during upload', () => {
    uploadTask.on.mockImplementation((state, observer, error, callback) =>
      Promise.resolve(observer())
    );

    return wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => {
        expect(uploadFile).toHaveBeenCalledTimes(1);
        expect(uploadFile).toHaveBeenLastCalledWith(files[0]);

        expect(wrapper).toMatchSnapshot();
      });
  });

  it('should update download url', () =>
    wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => {
        expect(uploadFile).toHaveBeenCalledTimes(1);
        expect(uploadFile).toHaveBeenLastCalledWith(files[0]);

        expect(getDownloadURL).toHaveBeenCalledTimes(1);

        return new Promise(resolve =>
          setTimeout(() => {
            expect(startUpdateDownloadURL).toHaveBeenCalledTimes(1);
            expect(startUpdateDownloadURL).toHaveBeenLastCalledWith(
              'www.download.com',
              'fileName'
            );
            resolve();
          }, 200)
        );
      }));

  it('should remove error, set uploading to false and reset input after upload', () =>
    wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => {
        expect(wrapper).toMatchSnapshot();

        expect(uploadFile).toHaveBeenCalledTimes(1);
        expect(uploadFile).toHaveBeenLastCalledWith(files[0]);

        new Promise(resolve =>
          setTimeout(() => {
            expect(wrapper).toMatchSnapshot();
            expect(submitEvent.target.reset).toHaveBeenCalledTimes(1);
            resolve();
          })
        );
      }));
});
