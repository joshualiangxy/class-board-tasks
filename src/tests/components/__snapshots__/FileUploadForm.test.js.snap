// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error should render cancelled error 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      File upload cancelled
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`error should render default error 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      File upload failed
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`error should render error when no file is selected 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      No file chosen
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`error should render unauthorized error 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      Not allowed to upload file!
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`file name should change file name back to no file chosen if no file selected 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      fileName
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`file name should change file name back to no file chosen if no file selected 2`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      No file chosen
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`file name should change file name on selecting file 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      fileName
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`render should render FileUploadForm 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={false}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      No file chosen
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={false}
    onClick={[Function]}
  >
    Upload
  </button>
</form>
`;

exports[`upload form should change button to "Uploading..." during upload 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={true}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      No file chosen
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={true}
    onClick={[Function]}
  >
    Uploading...
  </button>
</form>
`;

exports[`upload form should remove error, set uploading to false and reset input after upload 1`] = `
<form
  className="file-upload-form"
  onSubmit={[Function]}
>
  <input
    accept=".pdf"
    disabled={true}
    hidden="hidden"
    id="c1511c73-07b9-44cd-a4bd-01758515acb5"
    onChange={[Function]}
    onClick={[Function]}
    type="file"
  />
  <label
    className="button button--grey button--norm-grey"
    htmlFor="c1511c73-07b9-44cd-a4bd-01758515acb5"
  >
    Choose File
  </label>
  <div
    className="file-name"
  >
    <p>
      No file chosen
    </p>
  </div>
  <button
    className="button button--grey button--norm-grey"
    disabled={true}
    onClick={[Function]}
  >
    Uploading...
  </button>
</form>
`;
